packages/dds/tree/src/tree/treeTextFormat.ts

```diff
@@ -168,14 +169,33 @@ export function genericTreeKeys<T>(tree: GenericFieldsNode<T>): readonly FieldKe
         if (global === undefined) {
             return [];
         }
-        return (Object.getOwnPropertyNames(global) as GlobalFieldKey[]).map(symbolFromKey);
+        return (Object.getOwnPropertyNames(global) as GlobalFieldKey[])
+            .filter((key: GlobalFieldKey) => {
+                const value: Value = global[key];
+                return !Array.isArray(value) || value.length !== 0;
+            })
+            .map(symbolFromKey);
     }
+
+    // Omit any empty fields from local keys
+    const localKeys = (Object.getOwnPropertyNames(local) as LocalFieldKey[]).filter(
+        (key: LocalFieldKey) => {
+            const value: Value = local[key];
+            return !Array.isArray(value) || value.length !== 0;
+        },
+    );
+
     if (global === undefined) {
-        return Object.getOwnPropertyNames(local) as LocalFieldKey[];
+        return localKeys;
     }
     return [
-        ...(Object.getOwnPropertyNames(local) as LocalFieldKey[]),
-        ...(Object.getOwnPropertyNames(global) as GlobalFieldKey[]).map(symbolFromKey),
+        ...localKeys,
+        ...(Object.getOwnPropertyNames(global) as GlobalFieldKey[])
+            .filter((key: GlobalFieldKey) => {
```

Comment:
given that we have three copies of this filtering function, and it has a couple bugs, its probably better to split it out into its own faction and call in 3 times so we can maintain it in once place (if we want to keep it at all)